### PR TITLE
Fix links to uploaded files to use correct routes

### LINKED ISSUE #7205: Applicant PDF download after applying to a program with file upload question includes a link to download file that doesn't work
**Describe the bug**
When APPLICATION_EXPORTABLE flag is turned on (on the settings page), applicants are able to download a PDF of their application after submitting. If the program includes a file upload question, there will be a link to download the file in the PDF, but this doesn't work. We should change the behavior.

Note that the full link (https://staging-aws.civiform.dev/applicants/726268/files/applicant-726268%252Fprogram-108036%252Fblock-1%252Fcharlotte+%25281%2529+%25282%2529.jpg) will work if you're already logged in, but it won't otherwise. 

We'll plan to go with the option below, which involves downloading the uploaded files when exporting the application and changing the name in the export to be: `(PDF downloaded separately)` from `UPLOADED (click to download)` and removing the link property of the text

**To Reproduce**

1. Create an application with a file upload question
2. As an admin turn on APPLICATION_EXPORTABLE
3. Login as an applicant and fill out the application and export the PDF at the end
4. Click the download link in the PDF
5. Get navigated to home page

**Expected behavior**
Download the uploaded files when exporting the application and changing the name in the export to be: `(PDF downloaded separately)` from `UPLOADED (click to download)`. The text should no longer be clickable as well.

**Screenshots**
[name-unavailable (79548)-2024-04-11T08_37_10.198442.pdf](https://github.com/civiform/civiform/files/14948474/name-unavailable.79548.-2024-04-11T08_37_10.198442.pdf)

**Additional context**
See https://github.com/civiform/civiform/issues/7149 for further context

### PR DESCRIPTION
**Pull request by devin-civi**
### Description

This makes two main changes:

1) Change any links we expose to download files from the deprecated adminShow to acledAdminShow. This is no-op change since adminShow just redirects to acledAdminShow.
2) For PDF exports, if an applicant is exporting, use the applicant "show" route, rather than the admin show route. I think this was the primary cause of issue #7205.

Note if the applicant isn't logged in, the link will still fail. So there are probably more improvements we could do, but they're larger in scope. (Either prompting users for login for acl'd routes, or downloading all uploaded files along with the application, which is difficult because they'd need to be in a zip or some other package, which may be confusing for applicants.)

## Release notes

Fixed issue where applicant's could not download uploaded files from the PDF export of their application.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Issue(s) this completes

Fixes #7205

### CODE REVIEW

**Comment by devin-civi** [comment]
* **#7453** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7453?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @devin-civi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/services/applications/PdfExporterService.java [hunk 1]
```diff
@@ -28,10 +28,15 @@ public final class PdfExporterService {
    * to review applications.
    */
   public PdfExporter.InMemoryPdf generateApplicationPdf(
-      ApplicationModel application, boolean showEligibilityText, boolean includeHiddenBlocks) {
+      ApplicationModel application,
+      boolean showEligibilityText,
```
Comment (swatkat1) [inline server/app/services/applications/PdfExporterService.java:32]: Devin, showEligibilityText --> is also set for only admins. I think we can rename this variable to isAdmin and make it as one variable.

File: server/app/controllers/applicant/UpsellController.java [hunk 2]
```diff
@@ -213,7 +213,8 @@ public CompletionStage<Result> download(
                   pdfExporterService.generateApplicationPdf(
                       applicationMaybe.join().get(),
                       /* showEligibilityText= */ false,
-                      /* includeHiddenBlocks= */ false);
+                      /* includeHiddenBlocks= */ false,
```
Comment (swatkat1) [inline server/app/controllers/applicant/UpsellController.java:216]: Does include includeHiddenBlocks true only for admins? If yes, let us fold them too into into isAdmin

File: server/test/services/applications/PdfExporterServiceTest.java [hunk 3]
```diff
@@ -53,10 +53,7 @@ public void generateApplicationPdf() throws IOException {
       assertThat(AnnotationAction.get(PdfName.S)).isEqualTo(PdfName.URI);
       PdfString link = AnnotationAction.getAsString(PdfName.URI);
       assertThat(link.toString())
-          .isEqualTo(
-              "http://localhost:9000/admin/programs/"
-                  + applicationOne.getProgram().id
-                  + "/files/my-file-key");
+          .isEqualTo("http://localhost:9000/admin/applicant-files/my-file-key");
```
Comment (swatkat1) [inline server/test/services/applications/PdfExporterServiceTest.java:56]: So for acled file paths - no program id in the url. Am I going correctly?

**Review by swatkat1 — APPROVED** [review]
This is great!

File: server/test/services/applications/PdfExporterServiceTest.java (hunk 3 above)
Comment (devin-civi) [reply server/test/services/applications/PdfExporterServiceTest.java:56]: Yep. (You can see FileController for the methods that end up being called. The old adminShow takes a program ID, but doesn't use it and just calls into the new one. So all this change does is call directly into the new one, and thus doesn't need the program ID).

**Pull request merged**